### ISSUE TITLE
Issue with DL3008 and its documentation

### ISSUE THREAD

**Issue by eandersons** [hadolint/hadolint#999]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

1. Best Dockerfile practices link in [DL3008 page](/hadolint/hadolint/wiki/DL3008) points to <https://docs.docker.com/develop/develop-images/instructions/#apt-get>.
2. Either version pinning or cache busting with `apt-get update` is taken into account to mitigate outdated package retrieval from build cache, therefore DL3008 is not reported. 

   From [`apt-get` best Dockerfile practices instructions](https://docs.docker.com/develop/develop-images/instructions/#apt-get):
   
   > Using `RUN apt-get update && apt-get install -y` ensures your Dockerfile installs the latest package versions with no further coding or manual intervention. This technique is known as cache busting. You can also achieve cache busting by specifying a package version. This is known as version pinning.

   Taking this into account the current rationale and behaviour of DL3008 seems a bit outdated (examples in the aforementioned link confirms that).

   I am aware that version pinning still should be preferred, but in my opinion that should not be enforced or should be added as a separate rule that could be disabled when it would not be desirable.

### Actual behavior

1. Best Dockerfile practices link in [DL3008 page](/hadolint/hadolint/wiki/DL3008) points to <https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get>, where nothing is mentioned about `apt-get`.
2. Cache busting with `apt-get update` and `apt-get install` in the same `RUN` instruction currently is not taken into account thus reporting DL3008.

### Steps to reproduce the behavior

**Output of `hadolint --version`:** 

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
FROM gitpod/openvscode-server:1.85.1
RUN apt-get update && \
  apt-get --assume-yes install --no-install-recommends
    less \
    micro
```

**Additional environment details (OS, stack version, etc.)**

Ubuntu 22.04.3 LTS in a Docker container.

**Comment by GideonBear** [comment]
Ref #1164

**Comment by guest20** [comment]
> Best Dockerfile practices link in [DL3008 page](https://github.com/hadolint/hadolint/wiki/DL3008) points to https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get, where nothing is mentioned about apt-get.

That page has loads of things to say about `apt-get`, both on avoiding using outdated apt-caches because of layer caching and about specificity package versions in arguments.

For example: 
> Using apt-get update alone in a RUN statement causes caching issues and subsequent apt-get install instructions to fail. 

> Using RUN apt-get update && apt-get install -y --no-install-recommends ensures your Dockerfile installs the latest package versions

> Below is a well-formed RUN instruction that demonstrates all the apt-get recommendations.
